testing/trino-tests/src/test/java/io/trino/execution/TestSetSessionAuthorization.java

```diff
@@ -0,0 +1,258 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.execution;
+
+import com.google.common.collect.ImmutableMap;
+import io.airlift.units.Duration;
+import io.trino.client.ClientSession;
+import io.trino.client.StatementClient;
+import io.trino.plugin.base.security.FileBasedSystemAccessControl;
+import io.trino.spi.ErrorCode;
+import io.trino.spi.security.SystemAccessControl;
+import io.trino.testing.AbstractTestQueryFramework;
+import io.trino.testing.DistributedQueryRunner;
+import io.trino.testing.QueryRunner;
+import okhttp3.OkHttpClient;
+import org.testng.annotations.Test;
+
+import java.time.ZoneId;
+import java.util.Locale;
+import java.util.Optional;
+
+import static io.trino.SessionTestUtils.TEST_SESSION;
+import static io.trino.client.StatementClientFactory.newStatementClient;
+import static io.trino.spi.StandardErrorCode.GENERIC_USER_ERROR;
+import static io.trino.spi.StandardErrorCode.PERMISSION_DENIED;
+import static java.util.concurrent.TimeUnit.MINUTES;
+import static org.testng.Assert.assertEquals;
+
+public class TestSetSessionAuthorization
```

Comment:
@baohe-zhang can we add some tests for current_user and current_role in presence of authorization user?

Reply:
I think we are unable to do that. `select current_user` relies on the values of the server `Session` class. However, in the unit test, the DistributedQueryRunner's Session will not be updated by a previous sql query. It will use the initial default session for every new sql query.

Reply:
Figured out how to do that. We can directly pass authorizationUser to clientSession and execute select current_user with that clientSession. Added a unit test `testSelectCurrentUser` in TestSetSessionAuthorization class.